mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -311,24 +432,42 @@ class NaiveBayesModel private[ml] (
       require(value == 0.0 || value == 1.0,
         s"Bernoulli naive Bayes requires 0 or 1 feature values but found $features.")
     )
-    val prob = thetaMinusNegTheta.get.multiply(features)
+    val prob = thetaMinusNegTheta.multiply(features)
     BLAS.axpy(1.0, pi, prob)
-    BLAS.axpy(1.0, negThetaSum.get, prob)
+    BLAS.axpy(1.0, negThetaSum, prob)
     prob
   }
 
-  override protected def predictRaw(features: Vector): Vector = {
+  private def gaussianCalculation(features: Vector) = {
+    val prob = Array.ofDim[Double](numClasses)
+    var i = 0
+    while (i < numClasses) {
+      var s = 0.0
+      var j = 0
+      while (j < numFeatures) {
+        val d = features(j) - theta(i, j)
+        s += d * d / sigma(i, j)
+        j += 1
+      }
+      prob(i) = pi(i) - (s + logVarSum(i)) / 2
+      i += 1
+    }
+    Vectors.dense(prob)
+  }
+
+  @transient private lazy val predictRawFunc = {
```

Comment:
Does this need to be lazy or transient? it's just a function

Reply:
Oh it should be lazy, other it will cause:
```scala
java.util.NoSuchElementException: Failed to find a default value for modelType
[info]   at org.apache.spark.ml.param.Params.$anonfun$getOrDefault$2(params.scala:780)
[info]   at scala.Option.getOrElse(Option.scala:189)
[info]   at org.apache.spark.ml.param.Params.getOrDefault(params.scala:780)
[info]   at org.apache.spark.ml.param.Params.getOrDefault$(params.scala:777)
[info]   at org.apache.spark.ml.PipelineStage.getOrDefault(Pipeline.scala:43)
[info]   at org.apache.spark.ml.param.Params.$(params.scala:786)
[info]   at org.apache.spark.ml.param.Params.$$(params.scala:786)
[info]   at org.apache.spark.ml.PipelineStage.$(Pipeline.scala:43)
[info]   at org.apache.spark.ml.classification.NaiveBayesModel.<init>(NaiveBayes.scala:466)
```

Since  `NaiveBayesModel` should not contain `setDefault(modelType -> NaiveBayes.Multinomial)`

Reply:
I found that VectorIndexerModel also mark [transformFunc](https://github.com/apache/spark/blob/ed12b61784e2ce5a1779c162bde1e16e9a9a0135/mllib/src/main/scala/org/apache/spark/ml/feature/VectorIndexer.scala#L351) lazy.

Reply:
I mark it transient since I guess the precomputed matrices are included in this closure, whose size may be big in high-dim cases.